config/features/config.go

```diff
@@ -399,23 +440,63 @@ func logDisabled(flag cli.DocGenerationFlag) {
 
 // ValidateNetworkFlags validates provided flags and
 // prevents beacon node or validator to start
-// if more than one network flag is provided
+// if more than one network flag or if both old and new style flags are provided
 func ValidateNetworkFlags(ctx *cli.Context) error {
-	networkFlagsCount := 0
-	for _, flag := range NetworkFlags {
-		if ctx.IsSet(flag.Names()[0]) {
-			networkFlagsCount++
-			if networkFlagsCount > 1 {
-				// using a forLoop so future addition
-				// doesn't require changes in this function
-				var flagNames []string
-				for _, flag := range NetworkFlags {
-					flagNames = append(flagNames, "--"+flag.Names()[0])
-				}
-				return fmt.Errorf("cannot use more than one network flag at the same time. Possible network flags are: %s", strings.Join(flagNames, ", "))
-			}
+	// Check if the new network flag is provided
+	isNewNetworkFlagSet := ctx.IsSet(NetworkFlag.Names()[0])
+
+	// Count old-style network flags
+	oldNetworkFlagsCount := 0
+	oldNetworkFlagNames := []string{}
+
+	// Only counting explicitly set flags, ignoring defaults
+	if ctx.IsSet(Mainnet.Names()[0]) {
+		oldNetworkFlagsCount++
+		oldNetworkFlagNames = append(oldNetworkFlagNames, "--"+Mainnet.Names()[0])
+	}
+	if ctx.IsSet(SepoliaTestnet.Names()[0]) {
+		oldNetworkFlagsCount++
+		oldNetworkFlagNames = append(oldNetworkFlagNames, "--"+SepoliaTestnet.Names()[0])
+	}
+	if ctx.IsSet(HoleskyTestnet.Names()[0]) {
+		oldNetworkFlagsCount++
+		oldNetworkFlagNames = append(oldNetworkFlagNames, "--"+HoleskyTestnet.Names()[0])
+	}
+
+	// Error if both new and old style flags are used
+	if isNewNetworkFlagSet && oldNetworkFlagsCount > 0 {
+		return fmt.Errorf(
+			"cannot use both --network and old-style network flags at the same time. "+
+				"Please use only --network=%s instead of %s",
+			ctx.String(NetworkFlag.Names()[0]),
+			strings.Join(oldNetworkFlagNames, ", "),
+		)
+	}
+
+	// Error if multiple old-style network flags are provided
+	if oldNetworkFlagsCount > 1 {
+		return fmt.Errorf(
+			"cannot use more than one network flag at the same time. "+
+				"Consider using --network=<network> instead of %s",
+			strings.Join(oldNetworkFlagNames, ", "),
+		)
+	}
+
+	// Validate that network value is valid when using the new flag
+	if isNewNetworkFlagSet {
+		networkValue := ctx.String(NetworkFlag.Names()[0])
+		switch networkValue {
+		case "mainnet", "sepolia", "holesky":
+			// These are valid
+		default:
+			return fmt.Errorf(
+				"invalid network value %q for --network flag. "+
```

Comment:
```suggestion
				"invalid network value %s for --network flag. "+
```
`networkValue` is a single string

Reply:
%q enclose the network value in quotes, this could highlight issues like empty strings or unexpected spaces